### ISSUE TITLE
InternalTestAotSubset is failing in ILC x86 only

### ISSUE THREAD

**Issue by danmosemsft** [dotnet/corefx#21377]
```
Assert.Same() Failure\r\nExpected: \r\nActual:   
Stack Trace :
   at System.Tests.StringTests.InternalTestAotSubset()
   at xunit.console.netcore!<BaseAddress>+0xb808d1
   at SharedLibrary!<BaseAddress>+0x7626f3
   at SharedLibrary!<BaseAddress>+0x7624f3
```
@AtsushiKan this is your test. Seems 32 bit specific?

**Comment by morganbr** [comment]
@AtsushiKan , can you provide any context for this failure? I don't know anything about this test or failure...

**Comment by AtsushiKan** [comment]
Here's the test:

https://github.com/dotnet/corefx/blob/master/src/System.Runtime/tests/System/StringTests.cs#L2638

and the implementation of String.Intern (which should look familiar.)

https://github.com/dotnet/corert/blob/master/src/System.Private.CoreLib/src/System/String.Intern.cs#L12

We have two failures, both on RS3. (apparently not 32-bit specific.)

**Comment by AtsushiKan** [comment]
Ok, the trigger for causing the test failure is doing a multifile ILC build. Helix started running that way recently. The way to do that is to set EnableMultiFileILCTests=true or pass "-p:EnableMultiFileILCTests=true" when building the test project.

**Comment by AtsushiKan** [comment]
On multifile, "" is not the same reference as String.Empty. 

I guess that's not going to change...

**Comment by danmosemsft** [comment]
fixed by #21566

**Comment by danmosemsft** [comment]
fixed by #21566
